Adding tables modifiers (#300)
microsoft/atlas-design: site/src/components/table.md

				<td>Cell</td>
				<td>Cell</td>
			</tr>
			<tr>
				<td>Row three</td>
				<td>Cell</td>
				<td>Cell</td>
				<td>Cell</td>
				<td>Cell</td>
				<td>Cell</td>
				<td>Cell</td>
			</tr>
		</tbody>
	</table>
</div>

## Modifiers

All modifier classes should be applied in conjunction with the `.table` class.

### Simple

> _`table-simple`_ gives table has only horizontal cell borders.

Reviewer: I think we may want to tweak the writing a bit here:

```suggestion
_`table-simple`_ gives the table only horizontal cell borders.
```

Author: 🤦🏻‍♀️

Author: Updated

		<tr>
			<td>Row two</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row three</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row four</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row five</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
	</tbody>
</table>

### Size

_`table-sm`_/_`table-lg`_ makes table more or less condensed than the default.

<table class="table table-sm margin-top-sm">
	<thead>
		<tr>
			<th>Heading row</th>
			<th>Heading cell</th>
			<th>Heading cell</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>Row one</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row two</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row three</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
	</tbody>
</table>

<table class="table table-lg margin-top-sm">
	<thead>
		<tr>
			<th>Heading row</th>
			<th>Heading cell</th>
			<th>Heading cell</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>Row one</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row two</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row three</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
	</tbody>
</table>

### Stacked on mobile

> _`table-stacked-mobile`_ stacks table cells on mobile screen sizes.

Reviewer 2: On mobile, this doesn't look so good, I also don't see any code related to it.

![image](https://user-images.githubusercontent.com/30843002/148112564-d1dc15b2-4431-4243-9cea-aa4e0fae1efb.png)

Author: Ugh with a new default border around the table I forgot to take a closer look at this view. Good catch, updated!

## Modifiers

> _`table-simple`_ - simple table style only shows horizontal cell borders and no table border.

Reviewer 2: Thinking that this page has gotten extremely long. Just floating the idea, but perhaps we could just abbreviate or remove the code blocks for examples?

Author: I don't like how long the page became either, but I couldn't find out the best solution to shorten it. Removing all examples won't work, because I think some cases should be presented such as stacking on mobile, or how the complex table looks since it's pretty different from the default. And at the same time having examples for a few cases and not all feel weird to me. I'll try to shorten the code examples, showing only 2 rows in the table instead of 3 might work.

Reviewer 2: What about just removing the code blocks, not the examples?

Author: Updated

Reviewer 3: If we want to shorten the page, I'm wondering if the first example can be combined with the second example?

Author: I think it's fine the way it is right now after the latest update - having two code examples is much better than 9 as it used to be :)

Reviewer 2: I agree that the second table-wrapper example code seems like it could be removed.

Author: Updated

```html
<table class="table table-simple">
	<thead>
		<tr>
			<th>Heading row</th>
			<th>Heading cell</th>
			<th>Heading cell</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>Row one</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row two</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
		<tr>
			<td>Row three</td>
			<td>Cell</td>
			<td>Cell</td>
		</tr>
	</tbody>
</table>
```

> _`table-complex`_ - complex table style is a completely bordered cell layout with a heavy stylized header row.

Reviewer 2: Should we use headings?

Reviewer 2: ```md
### Complex
```

Author: Updated

		</tr>
	</tbody>
</table>

### Complex

> _`table-complex`_ - complex table style is a completely bordered cell layout with a heavy stylized header row.

Reviewer: I've got a little recommendation here to align how we talk about `table_complex` a bit more similarly to how we talk about the other classes:

```suggestion
_`table-complex`_ gives the table a completely bordered cell layout with a heavily stylized header row.
```

Author: Updated
